crates/iota-aws-orchestrator/src/orchestrator.rs

```diff
@@ -607,12 +607,67 @@ impl<P: ProtocolCommands<T> + ProtocolMetrics, T: BenchmarkType> Orchestrator<P,
             .iter()
             .collect();
         fs::create_dir_all(&path).expect("Failed to create log directory");
-        aggregator.save(path);
+        aggregator.save(&path);
+
+        if self.settings.enable_flamegraph {
+            self.fetch_flamegraphs(
+                parameters,
+                self.node_instances.clone(),
+                &path,
+                "?svg=true",
+                "flamegraph",
+            )
+            .await?;
+
+            if self
+                .settings
+                .build_configs
+                .get("iota-node")
+                .is_some_and(|config| config.features.iter().any(|f| f == "flamegraph-alloc"))
+            {
+                self.fetch_flamegraphs(
+                    parameters,
+                    self.node_instances.clone(),
```

Comment:
We could fetch for all nodes, including fullnodes via `self.instances_without_metrics()`